torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -836,8 +836,8 @@ class CudaKernelGenerator : private kir::IrVisitor {
              << "reduction::gridReduce<" << flags_str << ">(\n";
     indent() << kTab << gen(rop->out()) << ",\n";
     if (domain->hasBlockReduction()) {
-      indent() << kTab << "block_result"
-               << ",\n";
+      indent() << kTab << "block_result_" << block_reduce_name_ << ",\n";
+      block_reduce_name_++;
```

Comment:
nitpick/minor question: why do we want to increment the `block_reduce_name_` in `GridReduction` instead of doing it in code-gen of the block reduction itself? 

Also maybe just wondering if we'd ever want to change the ordering of block and grid reductions given that we now can have many grid reductions.

Reply:
From what I could tell the ordering when we have a block reduction followed by a grid reduction, is block reduction will be called first with the intermediate result going into "block_result...", then grid reduction will be called consuming that "block_result..." I couldn't find any instance where we had a block+grid reduction where we didn't first call the block reduction code generator then the grid reduction code generator. So I figured it was safest to increment in the grid reduction. It also seemed to me we would never generate "block_result..." without having a block -> grid reduction. So this seems safe to me.

Reply:
This should be safe, although I think ideally the dependency through the intermediate flag should be represented in KIR.  For now, the current method seems just fine.